envoy/common/backoff_strategy.h

```diff
@@ -28,6 +28,11 @@ class BackOffStrategy {
    * @param base_interval the new base interval for the backoff strategy.
    */
   virtual void reset(uint64_t base_interval) PURE;
+
+  /**
+   * @return if the time interval exceeds any configured cap on next backoff value.
+   */
+  virtual bool isOverTimeLimit(uint64_t /*interval_ms*/) const { return false; }
```

Comment:
I suspect I'm missing something obvious, but I don't think I understand we are adding this method to this interface? If we do need to add it, I'd suggest we make it PURE like the rest of the methods.